inspirehep/modules/orcid/cache.py

```diff
@@ -31,69 +31,114 @@
 from .exceptions import EmptyPutcodeError
 
 
+# Redis client as module-level singleton.
+_redis_client = None
+
+
+def _init_redis_client():
+    """
+    Note: the `_redis_client` module-level var cannot be initialized at module-load
+    time because accessing flask.current_app raises the exception:
+    "RuntimeError: Working outside of application context"
+    """
+    url = app.config.get('CACHE_REDIS_URL')
+    global _redis_client
+    _redis_client = StrictRedis.from_url(url)
+
+
 class OrcidCache(object):
     def __init__(self, orcid):
-        redis_url = app.config.get('CACHE_REDIS_URL')
-        self.redis = StrictRedis.from_url(redis_url)
+        """
+        Orcid cached data.
+
+        Args:
+            orcid (string): orcid identifier.
+        """
         self.orcid = orcid
+        self._cached_hash_value = None
+        self._new_hash_value = None
+        if not _redis_client:
+            _init_redis_client()
+        self.redis = _redis_client
 
-    def _get_key(self, recid):
+    def get_key(self, recid):
         """Return the string 'orcidcache:``orcid_value``:``recid``'"""
         return 'orcidcache:{}:{}'.format(self.orcid, recid)
 
-    def write_record_data(self, recid, putcode, hash_value=None):
+    def write_work_putcode(self, recid, putcode, inspire_record=None):
```

Comment:
this doesn't only store the putcode. What about `write_work_metadata`?

Reply:
I don't have a strong preference, but I want to keep the concept of hash unknown by the user of this class. Thus what this method does, is just writing the putcode.